### PR TITLE
Integration testing refactor and mesos bump

### PR DESCRIPTION
**Pull request by rveznaver**
This PR introduces the following:
- Update default Mesos version to 0.25.0
- Refactor testing suite
- Remove ZooKeeper and attribute testing for Mesos
- Remove support for Mesos < 0.23.0
- Set the default ```cgroups_hierarchy``` flag on the slave depending on the init system used (the Mesos 0.25.0 Linux launcher uses cgroups per default)
- Support for systemd on Debian and Ubuntu

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -1,112 +1,83 @@
...
+default['mesos']['version']    = '0.25.0'
 
 # Init system to use
-default['mesos']['init']    = case node['platform']
-                              when 'debian' then 'sysvinit_debian'
-                              when 'centos', 'amazon', 'scientific'
-                                node['platform_version'].to_i >= 7 ? 'systemd' : 'upstart'
-                              else 'upstart'
-                              end
+default['mesos']['init']       = case node['platform']
```
Comment (kamaradclimber) [inline attributes/default.rb]: FAR: use poise-service to avoid this

File: attributes/default.rb [hunk 2]
```diff
@@ -1,112 +1,83 @@
...
 
 #
 # Mesos MASTER configuration
 #
 
-# Mesos master binary location
-if node['mesos']['version'] == '0.21.1' || node['mesos']['version'] == '0.22.0' || node['mesos']['version'] == '0.22.1' || node['mesos']['version'] == '0.23.0'
-  default['mesos']['master']['bin']                         = '/usr/sbin/mesos-master'
-else
-  default['mesos']['master']['bin']                         = '/usr/local/sbin/mesos-master'
```
Comment (kamaradclimber) [inline attributes/default.rb]: you should mention in the release notes that older version of mesos support has been dropped

File: recipes/install.rb [hunk 3]
```diff
@@ -76,7 +76,7 @@
 template 'mesos-master-init' do
   case node['mesos']['init']
   when 'systemd'
-    path '/usr/lib/systemd/system/mesos-master.service'
+    path '/etc/systemd/system/mesos-master.service'
```
Comment (kamaradclimber) [inline recipes/install.rb]: +1

**Comment by kamaradclimber** [comment]
LGTM, that's especially good that you removed all the expectactions on the resource (pure duplicate of recipes with low value)

File: attributes/default.rb (hunk 2 above)
Comment (rveznaver) [inline attributes/default.rb]: Will do a pull request on the changelog and readme.

**Pull request merged**